orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -1502,5 +1502,45 @@ IMPLEMENT_GRADIENT_BUILDER(GetAbsGradient) {
   };
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetTileGradient) {
+  std::vector<NodeDef> result = {};
+  std::vector<Dimension> orig_shape,repeats_shape;
+
+  result.push_back(NodeDef("Shape", {I(0)}, {IA("orig_shape")})); // M,N,K
+  
+  std::vector<int64_t> axes_values = {1};
+  result.push_back(NodeDef("Unsqueeze", {IA("orig_shape")}, {IA("2d_orig_shape")}, {MakeAttribute("axes", axes_values)}));
+  result.push_back(NodeDef("Unsqueeze", {I(1)}, {IA("2d_repeats")}, {MakeAttribute("axes", axes_values)}));
+  result.push_back(NodeDef("Concat", {IA("2d_repeats"), IA("2d_orig_shape")}, {IA("concated_dims_T")},
+                           {MakeAttribute("axis", int64_t(1))}));  // [[a,M], [b,N], [c,K]]
+  
+  std::vector<int64_t> const_shape_minusone{-1};
+  NodeDef const_shape_minusone_node = ConstantVectorNode(const_shape_minusone, Name("const_shape_minusone"));
+  result.push_back(const_shape_minusone_node);
+  result.push_back(NodeDef("Reshape", {IA("concated_dims_T"),
+                            const_shape_minusone_node.output_args[0]},
+                            {IA("concated_dims_flatten")})); // flatten [a,M,b,N,c,K]
+
+  result.push_back(NodeDef("Reshape", {GO(0), IA("concated_dims_flatten")}, {IA("reshape_tile_grad_op")})); 
+
```

Comment:
we can have another optimized path here, 
if (the Rank of I(0) is known || Shape of I(1) is known) at gradient building time, we can construct the even indices as an initializer. 
You can use GetShape(). 

So we will be able to skip the Size and Range op.